runtime/core/exec_aten/util/tensor_util.h

```diff
@@ -484,6 +486,30 @@ inline bool tensor_is_type(
   return true;
 }
 
+inline bool tensor_is_type(
+    executorch::aten::Tensor t,
+    const std::vector<executorch::aten::ScalarType>& dtypes) {
+  if (std::find(dtypes.begin(), dtypes.end(), t.scalar_type()) !=
+      dtypes.end()) {
+    return true;
+  }
+
+  std::stringstream dtype_ss;
```

Comment:
It's not safe to assemble a dynamic string here. If you wanted to print all of the types, printing individual log lines would be the safest and easiest. If you wanted to assemble a string, you could create a fixed-size buffer on the stack and `strncat` strings onto it, but that can be error-prone.

Reply:
yes, I was thinking about using the c-ctyle buffer for that, I will update

Reply:
I decided to go a simple way that provides an explicit override without any loops. 

The use of strncat will lead to a warning https://gcc.gnu.org/bugzilla/show_bug.cgi?id=83404 which I think is not desired.

@dbort , hope you find that version suitable.

Reply:
@dbort surficing the thread on top.